### ISSUE TITLE
`smurf.set_targets()` only works on first call without reinitialization

### ISSUE THREAD

**Issue by BrianJKoopman** [simonsobs/sorunlib#55]
If you set the target pysmurf-controllers via `smurf.set_targets()`, then want to change to a different controller that isn't in the list you set the first time it won't work. For example:

```python
from sorunlib import *

initialize()  # creates CLIENT list with ['smurf1', 'smurf2'] in it
smurf.set_targets(['smurf1'])

# won't work
smurf.set_targets(['smurf2'])
```

This is because `smurf.set_targets()` merely selects items from the existing `run.CLIENTS` list. This means one has to re-`initialize()`. Perhaps `smurf.set_targets()` should re-build the `run.CLIENTS` list itself.

**Comment by guanyilun** [comment]
Since all initialize() does is to populate the client list, initialize is no longered needed, I guess

**Comment by BrianJKoopman** [comment]
> Since all initialize() does is to populate the client list, initialize is no longered needed, I guess

I don't think so, `smurf.set_targets()` will typically not be called at all, defaulting to using all available SMuRFs.

We still need to work out the details for how the SMuRF operations will be ordered (#54), but I don't expect to be constantly using `smurf.set_targets()` to select each slot.

**Comment by guanyilun** [comment]
I see, that makes sense!
